src/Microsoft.DotNet.Wpf/src/Shared/MS/Win32/SafeNativeMethodsOther.cs

```diff
@@ -224,11 +225,10 @@ private static partial class SafeNativeMethodsPrivate
             [DllImport(ExternDll.User32, SetLastError=true, CharSet=CharSet.Auto)]
             public static extern int GetCaretBlinkTime();
 
-            [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
-            public static extern bool GetStringTypeEx(uint locale, uint infoType, char[] sourceString, int count,
-                [Out, MarshalAs(UnmanagedType.LPArray, SizeParamIndex = 3)] UInt16[] charTypes);
+            [DllImport(ExternDll.Kernel32, SetLastError = true, CharSet = CharSet.Auto)]
+            public static unsafe extern bool GetStringTypeEx(uint locale, uint infoType, char* sourceString, int count, ushort* charTypes);
             
-            [DllImport(ExternDll.User32, CharSet = System.Runtime.InteropServices.CharSet.Auto)]
+            [DllImport(ExternDll.User32, CharSet = CharSet.Auto)]
```

Comment:
`CharSet` here can be ignore. But that's the logic of the old code.

Reply:
Yeah, you're right, but let's not do that here.